### DISCUSSION TITLE
Where is the source code for the downloads.docker.com website hosted?

### DISCUSSION THREAD

**Discussion by felipecrs** [moby/moby#47817] [Q&A] [answered]
Or is it open source at all? I am planning to ask Repology to track it, but they won't accept if it isn't open source.

**Comment by thaJeztah** [comment, marked as answer]
The site itself is an s3 bucket that's fronted by a CDN.

The deploy pipeline used to update the content of the bucket is in a private repository, and is responsible for signing the deb and rpm packages, repo-metadata, and generating HTML indexes.

So, as "website" goes, the only real part is generating the HTML indexes, of which an early version can be found in this repository (it's removed now, but older tags still contain it); https://github.com/moby/moby/blob/v1.13.1/hack/make/generate-index-listing

And this directory contains various scripts that were used at the time; https://github.com/moby/moby/tree/v1.13.1/hack/make, however a lot of moving parts changed (the website uses a rotating list of buckets which get synchronised, verified, and promoted to become the "live" version).

As to "packaging" itself, there's two repositories that are relevant;

- https://github.com/docker/docker-ce-packaging holds the scripts to build the deb and rpm packages for the docker engine, cli, compose, and buildx
- https://github.com/docker/containerd-packaging holds the same, but for the containerd.io packages

But there's a (still WIP) repository to replace both; https://github.com/docker/packaging

**Reply by felipecrs** [reply]
Sounds great! Thanks a lot for the elaborated response. I will let you know if I have news regarding Repology.
